spring-boot-starters/spring-boot-starter-data-geode/pom.xml

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starters</artifactId>
+		<version>1.4.0.BUILD-SNAPSHOT</version>
+	</parent>
+	<artifactId>spring-boot-starter-data-geode</artifactId>
+	<name>Spring Boot Data Geode Starter</name>
+	<description>
+		The Spring Boot Data Geode Starter including Spring Data GemFire with support for Apache Geode.
+	</description>
+	<url>http://projects.spring.io/spring-boot/</url>
+	<organization>
+		<name>Pivotal Software, Inc.</name>
+		<url>http://www.spring.io</url>
+	</organization>
+	<properties>
+		<main.basedir>${basedir}/../..</main.basedir>
+	</properties>
+	<dependencies>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.data</groupId>
+			<artifactId>spring-data-gemfire</artifactId>
+			<version>${spring-data-geode.version}</version>
```

Comment:
Your confusion is understandable.

Currently, if a user wants to include _Spring_ support for _Apache Geode_, then the user only needs to switch the version of _Spring Data GemFire_ that she is using.

For instance, she might be using _Spring Data GemFire_ for _Pivotal GemFire_, at version **1.8.0.RC1** (based on _GemFire 8.2_).  Or, perhaps she is using a release version of SDG, **1.7.4.RELEASE** (based on _GemFire 8.1_).  Either way, she may later decide to kick the tires on _Apache Geode_.  If so, she only need switch the version of the _Spring Data GemFire_ dependency to **1.7.0.APACHE-GEODE-EA-M1** in her application POM file (perhaps with a profile).

In hindsight, this may or may not have been the best idea, but several factors played into the decision...

1. The `apache-geode` branch was supposed to be short-lived as _Pivotal GemFire_ and _Apache Geode_ converged.  However, since **April 2015** they continue to diverge and the disparity between the 2 codebase is only growing.  For instance, you cannot compile SDG with both _Pivotal GemFire_ and _Apache Geode_ interchangeably, hence the separate branch.

2.  The former was mostly due to the features in GemFire they decided withhold from Geode (e.g. CQs, WAN) to distinguish the open source and commercial offerings.  Now they are throwing everything into the open source domain (eventually).  But, even with that, the package structure is set to change in _Apache Geode_ from `com.gemstone.gemfire` to `org.apache.geode`.  At some point, when/if _Pivotal GemFire_ converges, it's package structure will need to change as well.  However, this is a significant undertaking and not without customer impact.  Clearly, both of these affect SDG right now.

3.  Even when/if _Pivotal GemFire_ and _Apache Geode_ converge, they are likely to always have a separate set of dependencies.

4.   At the time, _Apache Geode_ had no official release and so did not even have a Maven artifact.  Now that _Apache Geode_ `1.0.0-incubating.M1` has been released and is also available in Maven Central, it is easier to have fixed releases of SDG based on these _Apache Geode_ versions (M2 release is planned for March 25th).

5. etc...

Which leads me to (as I mentioned previously in my initial comment)...

It maybe necessary now to have a separate SD project module for _Spring Data GemFire_ with _Apache Geode_ support.  I.e. let's create a **Spring Data Geode** project/module (which does NOT currently exist).

This new SD module may or may not be included in the SD release train.  Technically, it actually makes more sense if the **new** _Spring Data Geode_ module releases with _Apache Geode_, at least until Geode reaches GA.  FTR, SD release trains are not tied to either _Pivotal GemFire_ or _Apache Geode_ releases (as with any other SD driver, e.g. MongoDB, for that matter), nor should it.

Regardless, if SDG with _Apache Geode_ support were a separate SD module (i.e. **Spring Data Geode**), then it would not be necessary to override the version of SDG brought in by the SD BOM POM.

However, my thinking here was that, technically, a user can override any dependency version specified by a particular _Spring Boot_ version (e.g. `1.3.3.RELEASE`) containing the "blessed", "curated" set of dependencies as defined in the `[spring-boot-depedencies/pom.xml](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-dependencies/pom.xml#L45-L171)` file simply by **overriding** the desired dependency's version property in the user's application POM file, like so...

NOTE: by way of example, I will pick on "_GemFire_", which is [at **8.1.0** in _Spring Boot_ **1.3.3.RELEASE**](https://github.com/spring-projects/spring-boot/blob/v1.3.3.RELEASE/spring-boot-dependencies/pom.xml#L71))...

```
// application Maven pom.xml
<project ...>
  <modelVersion>4.0.0</modelVersion>
  <parent>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-parent</artifactId>
    <version>1.3.3.RELEASE</version>
  </parent>
  <groupId>com.example</groupId>
  <artifactId>example-spring-boot-gemfire-app</artifactId>
...
  <properties>
    ...
    <gemfire.version>8.2.0</gemfire.version>
  </properties>
...
</project>
```

Even though _Spring Boot_ **1.3.3.RELEASE** is based on _GemFire_ **8.1**, a user can effectively "override" that version and use, say, _GemFire_ **8.2**, simply by redeclaring the "`<gemfire.version>`" property in her application POM file.  This is true for nearly all dependencies (except for the SD BOM POM; explanation below).

So, following this thinking, my idea was that the _Spring Boot_ "Starter" for _Apache Geode_ could provide this "override" automatically for the user and include the right version of _Spring Data GemFire_ that supports _Apache Geode_ for development purposes simply by including the `spring-boot-starter-data-geode`.  Of course, this "override" would be approved/blessed by us for the user's convenience.

Except for the fact (and the "explanation") the SD BOM POM does not use dependency properties (e.g. `<spring-data-gemfire.version>`) when declaring the SD module versions in a particular BOM/release train (e.g. Hopper), and therefore, the SDG version is not separately overridable, as may be desired by the user.

So, how is a user to experiment with say another version of SD GemFire, or Redis, or Mongo, and so on, perhaps an experimental version containing a particular feature, in the context of _Spring Boot_ until the version of that SD module is included in the SD release train (BOM POM) and that release train version has been updated by the user manually ([here](https://github.com/spring-projects/spring-boot/blob/v1.4.0.M1/spring-boot-dependencies/pom.xml#L135)), or by us in a subsequent _Spring Boot_ release.  Make sense?

As one example, consider @christophstrobl  built SD _Repository_ support for SD Redis on a separate branch (e.g. `issues/DATAREDIS-425`), which had build snapshots ([1.7.0.DATAREDIS-425-SNAPSHOT](https://repo.spring.io/webapp/#/artifacts/browse/tree/General/libs-snapshot-local/org/springframework/data/spring-data-redis/1.7.0.DATAREDIS-425-SNAPSHOT)).

A user would be able to try this feature out (which was not technically [released until Hopper-RC1, or rather SD Redis 1.7.0.RC1](http://docs.spring.io/spring-data/redis/docs/1.7.0.RC1/changelog.txt)) in _Spring Boot_ **1.4.0.M1**, or event **1.3.3.RELEASE** (which was well **before** the `Hopper-RC1` release), simply by just overriding, say a "`<spring-data-redis.version>`" property, had it existed.

Anyway, I think I am tempted to start the `spring-data-geode` discussion now with the SD team.

Let me know if you have additional thoughts or concerns.  I mainly wanted to share my thinking here, and I am also prepared to do whatever the _Spring Boot_ team feels is most appropriate.

Thanks.